dadguide/models/monster_stats.py

```diff
@@ -41,24 +41,45 @@ class MonsterStats:
     PLUS_DICT = {'hp': 10, 'atk': 5, 'rcv': 3}
     LV_120_MULT_DICT = {'hp': 10, 'atk': 5, 'rcv': 5}
 
-    def stat(self, monster_model, key: StatType, lv, plus=99, inherit=False, is_plus_297=True,
-             stat_latents: MonsterStatModifierInput = None):
+    def stat(self, monster_model, key: StatType, lv, inherit_monster_model=None, plus=99, inherit=False, is_plus_297=True,
+             stat_latents: MonsterStatModifierInput = None, multiplayer=False):
+        # TODO: deal with atk-, rcv-, and hp- awakenings
         s_val = self.base_stat(key, lv, monster_model)
-
         if stat_latents and not (monster_model.is_equip or inherit):
             latents = s_val * stat_latents.get_latent_multiplier(key)
             stat_awakenings = stat_latents.get_awakening_addition(key)
             voice = s_val * stat_latents.num_voice_awakening * 0.1
             s_val += latents + stat_awakenings + voice
-
+        
         # include plus calculations. todo: is there a way to do this without subtraction?
         s_val += self.PLUS_DICT[key] * max(min(plus, 99), 0)
         if inherit:
             inherit_dict = {'hp': 0.10, 'atk': 0.05, 'rcv': 0.15}
             if not is_plus_297:
                 s_val -= self.PLUS_DICT[key] * max(min(plus, 99), 0)
             s_val *= inherit_dict[key]
-        return int(round(s_val))
+        
+        if inherit_monster_model:
+            # recursion is not possible because inherits do not have inherits on top of them
+            if monster_model.attr1.value == inherit_monster_model.attr1.value:
+                # add base stats of inherit only if main atts are the same
+                s_val += self.stat(inherit_monster_model, key, 99, inherit=True, multiplayer=False)
+
+            inherit_bonus = MonsterStatModifierInput(
+                num_hp_awakening=inherit_monster_model.awakening_count(1),
```

Comment:
can you import AwokenSkills enum from token_mappings.py and count AwokenSkills.ENHANCEDHP.value